app/models/concerns/hyku_addons/account_behavior.rb

```diff
@@ -83,5 +102,12 @@ def remove_settings_hash_key_with_nil_value
           settings[key].delete_if { |_hash_key, value| value.blank? } if settings[key].class == Hash
         end
       end
+
+      def map_array_fields
```

Comment:
This seems more like a controller or form concern to me than a model concern.  What do you think?

Reply:
Yeah let me try moving it to the controller